docs/source/routing/observability/router-telemetry-otel/apm-guides/datadog/router-instrumentation.mdx

```diff
@@ -0,0 +1,233 @@
+
+      subgraph:
+        attributes:
+          otel.name: subgraph
+          operation.name: "subgraph"
+          resource.name:
+            subgraph_operation_name: string
+```
+
+<Caution>
+High Cardinality: The `operation_name` and `subgraph_operation_name` attributes can be unbounded and high-cardinality if your GraphQL operations have many unique operation names. This affects both APM views and trace metrics, because Datadog creates metrics for each unique `resource.name` value. Consider using a more bounded attribute like `operation_type` (query/mutation/subscription) or removing the `resource.name` attribute and letting Datadog calculate the resource name on its own if you experience cardinality issues in Datadog.
+</Caution>
+
+With these attributes configured, you can filter for operations in Datadog APM:
+
+<img
+  className="screenshot"
+  alt="Datadog APM showing operations set with example attributes set in router.yaml"
+  src="../../../../../images/router/datadog-apm-ops-example.png"
+  width="600"
+/>
+
+## Error tracking
+
+Configure the `error.message` attribute to surface GraphQL errors and properly reflect them in Datadog APM.
+
+<Note>
+
+The error tracking configuration depends on the structure of your GraphQL error responses. The example below assumes errors are returned as an array with a `message` field. Adjust the JSONPath expression (`$[0].message`) to match your specific error response format.
+
+</Note>
+
+```yaml title="router.yaml"
+telemetry:
+  instrumentation:
+    spans:
+      supergraph:
+        attributes:
+          # Mark span as error when GraphQL errors occur
+          otel.status_code:
+            static: ERROR
+            condition:
+              eq:
+                - true
+                - on_graphql_error: true
+          # Capture the error message from the first error in the array
+          # Adjust the JSONPath to match your error response structure
+          error.message:
+            response_errors: $[0].message
+
+      subgraph:
+        attributes:
+          otel.status_code:
+            static: ERROR
+            condition:
+              eq:
+                - true
+                - subgraph_on_graphql_error: true
+          error.message:
+            subgraph_response_errors: $[0].message
+```
+
+## Metrics instrumentation
+
+Add GraphQL error tracking to your metrics for more insights into your GraphQL errors and correlate them with your supergraph and subgraph spans.
+
+```yaml title="router.yaml"
+telemetry:
+  instrumentation:
+    instruments:
+      router:
+        http.server.request.duration:
+          attributes:
+            # Track GraphQL errors in metrics
+            graphql.errors:
+              on_graphql_error: true
+
+      subgraph:
+        http.client.request.duration:
+          attributes:
+            subgraph.name: true
+            graphql.errors:
+              subgraph_on_graphql_error: true
+```
+
+## Complete configuration example
```

Comment:
What makes this "complete" compared to the basic instrumentation? When would we recommend this config over the other one? I assume that this config would be better than the basic one, so maybe we just start off with this one right away?